java-frontend/src/main/java/org/sonar/java/se/symbolicvalues/SymbolicValue.java

```diff
@@ -191,20 +193,33 @@ public void computedFrom(List<SymbolicValue> symbolicValues) {
   }
 
   public static class ExceptionalSymbolicValue extends SymbolicValue {
+    @Nullable
     private final Type exceptionType;
 
-    public ExceptionalSymbolicValue(int id, Type exceptionType) {
+    public ExceptionalSymbolicValue(int id, @Nullable Type exceptionType) {
       super(id);
       this.exceptionType = exceptionType;
     }
 
+    @CheckForNull
     public Type exceptionType() {
       return exceptionType;
     }
 
     @Override
     public String toString() {
-      return super.toString() + "_" + exceptionType.fullyQualifiedName() + "!";
+      return super.toString() + "_" + (exceptionType == null ? "!unknownException" : exceptionType.fullyQualifiedName()) + "!";
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) This class overrides "equals()" and should therefore also override "hashCode()". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1206)